batcher/aligned-sdk/src/communication/messaging.rs

```diff
@@ -190,9 +211,13 @@ async fn handle_batcher_response(msg: Message) -> Result<BatchInclusionData, Sub
             error!("Batcher responded with invalid max fee");
             Err(SubmitError::InvalidMaxFee)
         }
-        Ok(SubmitProofResponseMessage::InsufficientBalance(addr)) => {
+        Ok(SubmitProofResponseMessage::InsufficientBalance(addr, last_sent_valid_nonce)) => {
+            // If we receive an invalid balance we should grab the last_sent_valid_nonce.
             error!("Batcher responded with insufficient balance");
-            Err(SubmitError::InsufficientBalance(addr))
+            Err(SubmitError::InsufficientBalance(
+                addr,
+                last_sent_valid_nonce,
+            ))
         }
```

Comment:
i believe `last_sent_valid_nonce` was changed to `errored_nonce` (when removing the -1)